test/suite/test_compact11.py

```diff
@@ -0,0 +1,141 @@
+        except WiredTigerError as e:
+            if wiredtiger_strerror(WT_ROLLBACK) in str(e):
+                err = WT_ROLLBACK
+            elif wiredtiger_strerror(WT_PREPARE_CONFLICT) in str(e):
+                err = WT_PREPARE_CONFLICT
+            else:
+                raise e
+        lo_cursor.close()
+        hi_cursor.close()
+        self.assertEqual(err, 0)
+
+    def populate(self, uri, num_keys):
+        c = self.session.open_cursor(uri, None)
+        for k in range(num_keys // 10 * 9):
+            self.session.begin_transaction()
+            c[k] = ('%07d' % k) + '_' + 'aaaa' * ((self.value_size // 4) - 2)
+            self.session.commit_transaction(f'commit_timestamp={self.timestamp_str(2)}')
+        
+        for k in range(num_keys // 10 * 9, num_keys):
+            self.session.begin_transaction()
+            c[k] = ('%07d' % k) + '_' + 'bbbb' * ((self.value_size // 4) - 2)
+            self.session.commit_transaction(f'commit_timestamp={self.timestamp_str(2)}')
+        c.close()
+
+    def test_compact11_truncate(self):
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(1)}')
+
+        # Create and populate a table.
+        uri = self.uri_prefix
+        self.session.create(uri, self.create_params)
+
+        self.populate(uri, self.table_numkv)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Remove 1/4 of the keys at timestamp 3.
+        c = self.session.open_cursor(uri, None)
+        for i in range(self.table_numkv // 4):
+            self.session.begin_transaction()
+            c.set_key(i)
+            c.remove()
+            self.session.commit_transaction(f'commit_timestamp={self.timestamp_str(3)}')
+        c.close()
+
+        self.session.checkpoint()
+
+        # Move the oldest timestamp forward to make the previously removed keys obsolete.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(4)}')
+
+        # Fast truncate the last tenth of a file at timestamp 5.
+        self.session.begin_transaction()
+        self.truncate(uri, self.table_numkv // 10 * 9, self.table_numkv)
+        self.session.commit_transaction(f'commit_timestamp={self.timestamp_str(5)}')
+
+        self.session.checkpoint()
+
+        # Check the size of the table.
+        size_before_compact = self.get_size(uri)
+        
+        self.session.compact(uri, 'free_space_target=1MB')
+
+        # Ensure compact has moved the fast truncated pages at the end of the file.
+        # We should have recovered around 1/4 of the file.
+        size_after_compact = self.get_size(uri)
+        space_recovered = size_before_compact - size_after_compact
+        self.assertGreater(space_recovered, size_before_compact // 4)
+        
+        # Keep compact verbose messages on for debugging.
```

Comment:
I am not sure this comment is correct unless you meant we can remove those lines for easier debugging?

Reply:
How about `Ignore compact verbose messages used for debugging.`